### PR TITLE
KOGITO-2720 Apps and Examples checked in parallel

### PR DESCRIPTION
**Pull request by radtriste**
https://issues.redhat.com/browse/KOGITO-2720

Many thanks for submitting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

- [x] You have read the [contributors guide](CONTRIBUTING.md)
- [x] Pull Request title is properly formatted: `KOGITO-XYZ Subject`
- [ ] Pull Request title contains the target branch if not targeting master: `[0.9.x] KOGITO-XYZ Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains link to any dependent or related Pull Request
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -41,29 +41,33 @@ pipeline {
...
+        stage('Build apps and examples') {
+            parallel {
+                stage('Build integration-tests with persistence') {
+                    steps {
+                        mavenCleanInstall("integration-tests", false, ["persistence"])
+                    }
+                }
+                stage('Build kogito-apps') {
+                    steps {
+                        mavenCleanInstall("kogito-apps")
```
Comment (danielezonca) [inline Jenkinsfile:53]: Are you sure it is possible to build apps and example in parallel?
As far as I remember some examples are using data-index and probably other so it should not work.
Can you please verify?

Comment (cristianonicolai) [reply Jenkinsfile:53]: it should be possible now.... the dependency for kogito-apps in the examples should be at runtime only. Any integration/test should be in kogito-apps itself as @tiagodolphine showed.
of course there might be a challange to get things stable in regards to random ports/conflicts

Comment (radtriste) [reply Jenkinsfile:53]: issue is part of epic https://issues.redhat.com/browse/KOGITO-2715 and it was decided to go for that way. but indeed, there may be some work to be done on port conflicts ...

Comment (radtriste) [reply Jenkinsfile:53]: not sure I have the time though to handle the port conflicts ... (leaving on holidays on Friday)

**Comment by radtriste** [comment]
Jenkins retest this

File: Jenkinsfile (hunk 1 above)
Comment (danielezonca) [reply Jenkinsfile:53]: Actually I was thinking at test automation for some end to end example like travel agency. Btw you are probably right that now with integration tests module in kogito-apps we can verify the integration there :+1:

**Comment by radtriste** [comment]
Almost there but getting `java.io.FileNotFoundException: Too many open files` :(

**Comment by radtriste** [comment]
Jenkins retest this

**Comment by radtriste** [comment]
The parallel checks are taking much more time than sequential. It won't be improvement unless we really raise the memory capacity (which is already 16 GB ...)

**Comment by danielezonca** [comment]
Ok last build took almost than 4h 30m to run and 1 test failed too so we have not enough resources to run in parallel.
I think it is better to close this PR and stay with the current setup. Wdyt?

**Comment by Sgitario** [comment]
> Ok last build took almost than 4h 30m to run and 1 test failed too so we have not enough resources to run in parallel.
> I think it is better to close this PR and stay with the current setup. Wdyt?

I have fixed the test failing in https://github.com/kiegroup/kogito-examples/pull/348. 
About the build time, I will try to do some improvements and if it's not better than 2h 30m, I think it makes sense to close this PR.

**Comment by Sgitario** [comment]
Let's close this PR as no improvements have been achieved.

**Pull request closed without merging**